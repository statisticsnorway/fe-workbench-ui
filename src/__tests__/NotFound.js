import React from 'react'
import { shallow } from 'enzyme'

import NotFound from '../pages/404/NotFound'

describe('NotFound', () => {
  it('Renders itself without error and only once', () => {
    const properties = {
      location: '/something/that/does/not/exist',
      languageCode: 'en'
    }

    const component = shallow(<NotFound {...properties} />)

    expect(component.length).toEqual(1)
    expect(component).toMatchSnapshot()
  })
})
