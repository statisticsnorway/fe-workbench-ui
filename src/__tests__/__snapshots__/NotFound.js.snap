// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Renders itself without error and only once 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFound
    languageCode="en"
    location="/something/that/does/not/exist"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Icon
          as="i"
          circular={true}
          color="red"
          name="eye slash outline"
          size="massive"
        />,
        <Header
          as="h1"
          content="Sorry :'("
          subheader="Could not find anything located at 'undefined'"
        />,
      ],
      "textAlign": "center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "i",
          "circular": true,
          "color": "red",
          "name": "eye slash outline",
          "size": "massive",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h1",
          "content": "Sorry :'(",
          "subheader": "Could not find anything located at 'undefined'",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Icon
            as="i"
            circular={true}
            color="red"
            name="eye slash outline"
            size="massive"
          />,
          <Header
            as="h1"
            content="Sorry :'("
            subheader="Could not find anything located at 'undefined'"
          />,
        ],
        "textAlign": "center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "circular": true,
            "color": "red",
            "name": "eye slash outline",
            "size": "massive",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h1",
            "content": "Sorry :'(",
            "subheader": "Could not find anything located at 'undefined'",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
